tests/combine_samples.C

```diff
@@ -0,0 +1,190 @@
+            } 
+            else if (offset_arg) {
+                offset_file = argv[i];
+                offset_arg = false;
+            }
+            else if (!strcmp(argv[i], "offset") || !strcmp(argv[i], "-o")) {
+                if (rank==0) std::cout << "Will subtract offset" << std::endl;
+                subtract_offset = true;
+                offset_arg = true;
+            }
+            else if (!strcmp(argv[i], "mean") || !strcmp(argv[i], "-m")) {
+                if (rank==0) std::cout << "Will subtract mean" << std::endl;
+	    	subtract_mean = true;
+	    }
+	    else {
+                sample_names.push_back(argv[i]); 
+	    }
+	}
+    }
+    else {
+        if (rank==0) std::cout << "No arguments passed." << std::endl;
+        return 1;
+    }
+	
+    CAROM_VERIFY(!(subtract_mean && subtract_offset)); // both should not be selected 
+ 
+    /*-- Read dimension and count number of snapshots/bases --*/
+    if (rank==0) std::cout << "Opening files to read dimension and count number of snapshots/bases" << std::endl;
+    int dimFirst = 0;
+    for (const auto& sample_name: sample_names) {
+        CAROM::BasisReader reader(sample_name);
+		
+	    dim    = reader.getDim(kind, 0);
+ 	    snaps += reader.getNumSamples(kind, 0);
+	    if (dimFirst == 0) dimFirst = dim;
+	    
+	    CAROM_VERIFY(dim == dimFirst); // ensures all files have the same dimension distribution
+    }
+	
+    CAROM_VERIFY((snaps > 0) && (dim > 0));
+	
+    /*-- Load data from input files --*/
+    std::string generator_filename = "total";
+    std::unique_ptr<CAROM::BasisGenerator> static_basis_generator;
+    static_basis_generator.reset(new CAROM::BasisGenerator(
+                                     CAROM::Options(dim, snaps).setMaxBasisDimension(snaps), false,
+                                     generator_filename));
+
+    if (rank==0) std::cout << "Loading data from " << kind << std::endl;
+    for(const auto& sample_name: sample_names) {
+        static_basis_generator->loadSamples(sample_name, kind);
+    }
+
+    if (rank==0) std::cout << "Saving data uploaded as a snapshot matrix" << std::endl;
+    static_basis_generator->writeSnapshot();
+	
+    if (!subtract_mean && !subtract_offset) {
+        /*-- Compute SVD and save file --*/
+        if (rank==0) std::cout << "Computing SVD" << std::endl;
+        int rom_dim = static_basis_generator->getSpatialBasis()->numColumns();
+        if (rank==0) std::cout << "U ROM Dimension: " << rom_dim << std::endl;
+        static_basis_generator->endSamples();
+    }
+    else {
+    	/*-- load data from hdf5 file to find the mean and subtract it --*/
+    	if (rank==0) std::cout << "Reading snapshots" << std::endl;
+    	CAROM::Matrix *snapshots = (CAROM::Matrix*) static_basis_generator->getSnapshotMatrix();
+        
+    	int num_rows = snapshots->numRows();
+    	int num_cols = snapshots->numColumns();
+        CAROM::Vector *offset = new CAROM::Vector(num_rows, true);
```

Comment:
This could be `CAROM::Vector offset(num_rows, true);` rather than using `new` and `delete`.